File: movies_picker_client/lib/data/models/location.dart
import '../../shared/shared.dart';

enum Location {
  bangalore,
  chennai,
  delhi,
  pune,
}

extension LocationExtension on Location {
  String get display => SharedFunctions.getEnumName(this);
}

abstract class LocationParser {
  static Location parse(String s) {
    switch (s) {
      case 'BANGALORE':
        return Location.bangalore;
      case 'CHENNAI':
        return Location.chennai;
      case 'DELHI':
        return Location.delhi;
      case 'PUNE':
      default:
        return Location.pune;
    }
  }
}

File: movies_picker_server/movies-picker/src/main/java/moviespicker/model/Location.java
package moviespicker.model;

public enum Location {
    BANGALORE,
    CHENNAI,
    DELHI,
    PUNE;

    public static Location parse(String s) {
        switch (s.toUpperCase()) {
            case "BANGALORE": return BANGALORE;
            case "CHENNAI": return CHENNAI;
            case "DELHI": return DELHI;
            case "PUNE":
            default:
            return PUNE;
        }
    }
}
